packages/contracts-bedrock/src/L1/LGEStaking.sol

```diff
@@ -0,0 +1,220 @@
+        totalDeposited[_token] -= _amount;
+        IERC20(_token).safeTransfer(msg.sender, _amount);
+        emit Withdraw(_token, _amount, msg.sender);
+    }
+
+    /// MIGRATE ///
+
+    /// @notice Migrates assets to the specified L2 destination.
+    /// @param  _l2Destination The address of the L2 destination to migrate tokens to.
+    /// @param  _tokens An array of token addresses to migrate.
+    /// @dev    Sends assets to the migration contract, and then calls `migrate` to move the assets.
+    function migrate(address _l2Destination, address[] calldata _tokens) external nonReentrant whenNotPaused {
+        require(migrationActivated(), "LGE Staking: Migration not active.");
+        require(_l2Destination != address(0), "LGE Staking: May not send tokens to the zero address.");
+        uint256 length = _tokens.length;
+        require(length > 0, "LGE Staking: Must migrate some tokens.");
+        uint256[] memory amounts = new uint256[](length);
+        uint256 amount;
+        for (uint256 i; i < length; i++) {
+            amount = balance[_tokens[i]][msg.sender];
+            require(amount > 0, "LGE Staking: No tokens to migrate.");
+            balance[_tokens[i]][msg.sender] -= amount;
+            totalDeposited[_tokens[i]] -= amount;
+            amounts[i] = amount;
+            IERC20(_tokens[i]).safeTransfer(address(lgeMigration), amount);
+        }
+        lgeMigration.migrate(_l2Destination, _tokens, amounts);
+        emit TokensMigrated(msg.sender, _l2Destination, _tokens, amounts);
+    }
+
+    /// OWNER ///
+
+    /// @notice This function allows the owner to either add or remove an allow-listed token for deposit.
+    /// @param  _token The token address to add or remove.
+    /// @param  _set A boolean for whether the token is allow-listed or not. True for allow-listed, false otherwise.
+    function setAllowlist(address _token, bool _set) external onlyOwner {
+        allowlisted[_token] = _set;
+        emit AllowlistSet(_token, _set);
+    }
+
+    /// @notice This function allows the owner to modify the deposit cap for deposited tokens.
+    /// @param  _token The token address to modify the deposit cap.
+    /// @param  _newDepositCap The new deposit cap.
+    function setDepositCap(address _token, uint256 _newDepositCap) external onlyOwner {
+        depositCap[_token] = _newDepositCap;
+        emit DepositCapSet(_token, _newDepositCap);
+    }
+
+    /// @notice This function allows the owner to set the migration contract used to move deposited assets to the
+    ///         Ozean L2.
+    /// @param  _contract The new contract address for the LGE Migration logic.
+    /// @dev    The new migration contract must conform to the ILGEMigration interface.
+    /// @dev    If this contract is set to address(0) migration is deactivated
+    function setMigrationContract(address _contract) external onlyOwner {
+        lgeMigration = ILGEMigration(_contract);
```

Comment:
if lgeMigration is set to malicious contract then users calling "migrate" will be transferred to it. 

Need to ensure Owner control is set to multisig